src/Data/CurrentUser/CurrentUserDataItem.ts

```diff
@@ -64,16 +68,23 @@ export const CurrentUser = provideDataItem('CurrentUser', {
   connection: {
     async get() {
       const user = await load(currentUser)
+
+      const Authorization = getTokenAuthorization()
+      if (Authorization) {
+        if (!user) return getTokenBasedCurrentUser(Authorization)
+        else verifySameWhoAmIUser(user.email(), Authorization)
```

Comment:
* [ ] FYI The fact that we're only checking this in the current user data item, and not in the GDPR data class, reinforces my previous feedback about an architectural issue. It would be better to move this check to the app boot process.